fontbe/src/features/marks.rs

```diff
@@ -305,6 +334,63 @@ impl<'a> MarkLookupBuilder<'a> {
         }
         result
     }
+
+    fn make_mark_to_liga_groups(&self) -> BTreeMap<GroupName, MarkGroup<'_>> {
+        let mut groups = BTreeMap::<_, MarkGroup>::new();
+        let mut liga_anchor_groups = HashSet::new();
+
+        // a temporary buffer, reused for each glyph
+        let mut component_groups = HashMap::new();
+        for (glyph_name, anchors) in &self.anchor_lists {
+            let is_not_liga = !self.gdef_classes.is_empty()
```

Comment:
negative booleans are a common source of error. Strongly recommend making it positive, e.g. is liga.

This loop is a bit hard to read due to all the conditions before we actually do anything. I wonder if some of the continue cases could be factored into a function so you could say something like `for (name, anchors, max_ligature_index) in self.mark_to_ligature_anchors()` would help?

Reply:
👍 on the boolean.

looked at splitting this fn but it feels like a marginal gain, so I've just improved the comments a bit to explain the flow.